### ISSUE TITLE
Sharing Settings 

### ISSUE THREAD

**Issue by josephdouce** [exodus4d/pathfinder#170]
The fact that I cant add out of Corp players to a Corp map is crushing this tool for me everything else is great. 

I think the you should implement tripwire like functionality. If you are logged in with a Corp Account then it tracks your location and you can see Corp Maps, even if the character your logged in as is not in corp. 

This allows many people to have there alts connected to the corp map without adding each alt individually. The only way we could get it set up as we want now would be to make a private map then invite everyone from Corp and there Alts 200+, even if we could add alts to the map would ahve to manually add 100 accounts... which is still impractical.

Alternatively, if I have 2 accounts logged in with EVE SSO, both characters should have access to all the maps.

**Comment by exodus4d** [comment]
I agree. This would be nice. There are so many ways people play eve, that
it is impossible to match everyone's needs. When I started this project, i
thought about just smaller groups that need easy Access to a Mapping Tool
without a complex Setup process.

Maybe we can implementiert something like "Group share" in future releases.

josephdouce <notifications@github.com> schrieb am Mi., 25. Mai 2016, 18:55:

**Comment by Tupsi** [comment]
This ties together with another suggestion here about securing the map against unwanted changes about who has access. Might be worth talking about it over in an extra slack channel for a later release?

**Comment by josephdouce** [comment]
Just have it so all maps are linked to the user not to the character.

**Comment by Betriebsrat** [comment]
+1 the disability to share alliance maps with out of alliance alts is currently what is us holding back using it

**Comment by dkrotil** [comment]
I think this is something very complex, because you need create some ACL system around Pathfinder.  Now you can control global access to corp or alliance with ID, this should be considered too. You will don´t use Pathfinder for long time, because I doubt, that this can be implemented quickly.

**Comment by CokkocZateki** [comment]
Could it not work off standings? +5 or +10

**Comment by dkrotil** [comment]
Maybe, this is @exodus4d call . You can build ACL around Pathfinder to be independent from, for example , standings in Eve or you can make tight bound between standings and ACL in Pathfinder, not sure if everybody likes that idea. 
But I don´t saying this bad idea.

**Comment by Betriebsrat** [comment]
you are able to share alliance maps with alliances, corp maps with corps and private maps with any char right now.
why is it difficult to expand the alliance map scope to any char and any corp?

**Comment by CokkocZateki** [comment]
But we need to be able to share any map with any alt. Would that be
plausible

**Comment by dkrotil** [comment]
@Betriebsrat and @CokkocZateki if you have an idea HOW to do it, then post it. Otherwise @exodus4d has some many work on Pathfinder , he have no time to read posts like this, which aren´t telling anything new. 
Yes I need to share maps with alts too. But I need security , if you don´t, then use Siggy or Tripwire.

**Comment by atkinsj** [comment]
It really isn't hard and I'm not sure why you're making it out that way.
Just have it so that corporation or alliance maps can have also have
characters specified on then (which is already implemented on the Private
map type).

To be quite honest in our version I've removed the concept of
private/Corp/alliance and on each map you can specify a mixture of all 3.
It's only a few lines of code change to allow it.

**Comment by Betriebsrat** [comment]
no clue why a few guys are on the edge of getting pissed off by that request.

thank you @atkinsj for an actual solution

**Comment by Betriebsrat** [comment]
@atkinsj could you provide me a diff or the actual changed files? tried it myself but no luck yet

**Comment by atkinsj** [comment]
@Betriebsrat I can take a look; we have a lot of modifications on our fork so trying to cherry pick these might be a challenge. You basically need to modify the template so that alliance/corporation/character show all the time instead of some of the time, modify the MapModel (getData, getActiveCharacters, and probably hasAccess which I've changed a lot so would need to check), and controller/api/map.php (the save function, just ctrl+f for 'isprivate').

In all of those cases you can see that setting map permissions is just an if/else chain. You can pretty much remove the if statements and look for valid corp/character/alliance permissions regardless.

I'll try to get to it in the next few days, though.

**Comment by josephdouce** [comment]
Link to the fork?

**Comment by flyingferret** [comment]
Yeah trying to convince people in my alliance to use pathfinder over tripwire and this topic keeps coming up at a no go.   any solutions yet ??
